src/controller/java/JniReferences.cpp

```diff
@@ -21,37 +21,54 @@
 
 #include <support/CodeUtils.h>
 
-pthread_mutex_t JniReferences::sStackLock  = PTHREAD_MUTEX_INITIALIZER;
-JavaVM * JniReferences::sJvm               = nullptr;
-jclass JniReferences::sClusterExceptionCls = nullptr;
+namespace chip {
+namespace Controller {
+namespace JniReferences {
 
-void JniReferences::SetJavaVm(JavaVM * jvm)
+pthread_mutex_t sStackLock = PTHREAD_MUTEX_INITIALIZER;
```

Comment:
Shouldn't JniReferences be a class instead of a namespace disguises as a class that uses a bunch of static variables?

Reply:
I can revert it, but I originally made this change because of a comment @andy31415 made in #7439. Andrei, any comment here?

Reply:
What about making sClassLoader and the other variables non-static data members of the class and actually creating an instance of the class? That would solve Andrei's objection that the class contained only static members.

This transformation made it less obvious that sClassLoader is only used in some of the functions. Namespaces don't have private fields, although moving those other functions into another file and putting sClassLoader et al into an unnamed namespace would fix that.

I am not sure if I agree with the objection to begin with, classes grouping related data does not seem useless even if everything is static because private visibility can be used in a class (again, separate files and unnamed spaces can achieve similar encapsulation). A namespace is more often very large and the same throughout an entire component.

Personally I am actually bothered that in CHIP namespaces use a mixture of lowercase and uppercase naming styles and I cannot tell if a scope is a class or a namespace. If I see ::chip::Controller::JniReferences::GetStackLock I would definitely assume that JniReferences is a class.

In the end, this is an issue of style and expectation and not functionality. I wouldn't reject a patch over it when we have no codified rules, but the new version of this code is actually violating more of my expectations than the previous version was.

Reply:
I would prefer a singleton class. Makes it clear that things are related and does not have a 'all static' methods/fields and only one thing is static.

Reply:
Made it a singleton. Thanks for the explanation!